### PR TITLE
Pt 167961470 stop off-chain updates on-chain Lima on

### PR DESCRIPTION
**Pull request by velzevur**
[SC protocol: stop off-chain updates on-chian since Lima](https://www.pivotaltracker.com/story/show/167961470)

This must be anounced loudly so all existing channels that are still using the old `aesc_offchain_tx` version either make an update round using the new version or either close the channel if the other party is not responding/cooperating.

This will allow for different off-chain protocols to be build without the current requirement their updates to be deserializable on-chain.

### CODE REVIEW

**Comment by velzevur** [comment]
Rebased on top of latest `master`

File: apps/aechannel/test/aesc_test_utils.erl [hunk 1]
```diff
@@ -395,23 +395,23 @@ state_tx(ChannelPubKey, Initiator, Responder, Spec0) ->
...
-            state_hash         => StateHash,
-            round              => maps:get(round, Spec)},
     OffChainSpec =
-        case maps:get(block_hash, Spec0, none) of
-            none -> OffChainSpec0;
-            Val -> OffChainSpec0#{block_hash => Val}
-        end,
+        lists:foldl(
+            fun(Key, AccumSpec) ->
+                case maps:get(Key, Spec0, none) of
```
Comment (uwiger) [inline apps/aechannel/test/aesc_test_utils.erl]: I find `maps:find/2` more intuitive in these cases.

Comment (uwiger) [reply apps/aechannel/test/aesc_test_utils.erl]: Perhaps even better would be `maps:merge(#{channel_id => ...}, maps:with([block_hash, updates], Spec0))`

**Comment by velzevur** [comment]
contract's test failed
```

%%% aehttp_contracts_SUITE ==> aevm.remote_gas_test_contract: FAILED
%%% aehttp_contracts_SUITE ==> {{badmatch,[{key_block,{key_header,144,
                                   <<183,151,150,159,253,163,226,70,120,59,47,
                                     208,211,235,137,46,65,179,82,21,5,238,211,
                                     49,3,136,14,118,104,98,203,37>>,
                                   <<183,151,150,159,253,163,226,70,120,59,47,
                                     208,211,235,137,46,65,179,82,21,5,238,211,
                                     49,3,136,14,118,104,98,203,37>>,
                                   <<31,116,235,202,70,80,62,80,120,51,193,203,
                                     11,15,101,65,143,209,28,92,119,156,195,
                                     253,57,193,248,224,4,1,74,99>>,
                                   553713663,7790315590093093865,
                                   1566392040507,2,
                                   [1337],
                                   <<227,121,219,113,221,53,6,7,252,223,57,46,
                                     113,39,116,29,43,62,0,214,4,180,158,177,
                                     249,87,117,234,31,120,183,24>>,
                                   <<217,202,108,173,192,99,13,10,129,124,71,
                                     86,232,121,148,177,243,254,160,88,174,204,
                                     22,114,15,42,51,71,75,19,135,16>>,
                                   <<0,0,0,1>>}}]},
 [{aecore_suite_utils,mine_blocks_until_txs_on_chain,5,
                      [{file,"/home/builder/aeternity/apps/aecore/test/aecore_suite_utils.erl"},
                       {line,322}]},
  {aehttp_contracts_SUITE,force_fun_calls,2,
                          [{file,"/home/builder/aeternity/apps/aehttp/test/aehttp_contracts_SUITE.erl"},
                           {line,1303}]},
  {aehttp_contracts_SUITE,remote_gas_test_contract,1,
                          [{file,"/home/builder/aeternity/apps/aehttp/test/aehttp_contracts_SUITE.erl"},
                           {line,1156}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1546}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1062}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,994}]}]}
.
```
It looks unrelated to this changes. Restarted the job

File: docs/release-notes/next-lima/PT-167961470_stop_offchain_updates_onchain_since_Lima.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+* Obsoletes the old State Channel off-chain transaction type which contained
+  the list of updates that produced the latest `state_hash`. The new
+  transaction type is available since Fortuna hard fork and the FSM is
+  producing such transactions ever since. This detaches the off-chain protocol
+  from the on-chain one and allows development of unique off-chain protocols
+  that don't need their updates to be serializable on-chain.
```
Comment (velzevur) [inline docs/release-notes/next-lima/PT-167961470_stop_offchain_updates_onchain_since_Lima.md]:
```suggestion
  that don't need their updates to be serializable on-chain. From Lima hard
  fork on, old version off-chain transactions will not be allowed as a payload
  in on-chain dispute transactions. For currently existing state channels which
  latest co-authenticated state is an old version off-chain transaction is
  suggested to make a new co-authenticated transaction that is using the new
  serialization. If the other party refuses to authenticate a new round or is
  simply missing, one can use the solo closing sequence instead.
```

**Pull request merged**